devenv/docker/blocks/prometheus_basic_auth_proxy/nginx.conf

```diff
@@ -0,0 +1,34 @@
+events { }
+
+http {
+  server {
+
+    listen 10090;
+
+    location / {
+
+      # Removes any Access-Control-Allow-Origin from Prometheus itself. When accessing from browser, having * or
+      # multiple values is not allowed in some cases
+      proxy_hide_header Access-Control-Allow-Origin;
+
+      # Allow the origin access. This is kinda wildcard but for browser it seems more strict and is needed for
+      # withCredentials requests.
+      add_header Access-Control-Allow-Origin $http_origin;
+
+      # When using withCredentials requests this must be true.
+      add_header Access-Control-Allow-Credentials true;
```

Comment:
This is a bit weird that this is needed and I think there could be a bug in the code. In multiple datasources we have this code:
```javascript
    if (this.basicAuth || this.withCredentials) {
      options.withCredentials = true;
    }
```
Which means we are doing `withCredentials` request if basic auth is set. It makes sense if it is not set as it will send cookies or cached basic auth header to the datasource from the browser so the auth basically happens outside of Grafana. If the basic auth is set in Grafana though we manually add the auth header and we do not rely on browser adding it automatically (I am actually not sure what happens if browser has some auth cached, which will be there in the end). So I think using withCredentials request if we have basic auth set is not necessary and then this config could be simplified but I am not 100% on that. So if anybody knows better let me know.

Reply:
@torkelo Can you help me with this?

Reply:
Not sure, but basicAuth is only true here if the access mode is Browser. So think with credentials needs to be true in those scenarios. 

Here is someone who needed this for Elasticsearch: https://github.com/grafana/grafana/pull/823

Reply:
> Not sure, but basicAuth is only true here if the access mode is Browser. So think with credentials needs to be true in those scenarios.

Why do you think it is needed? It seems to me that basic auth should work even without `withCredentials = true` and if you need credentials you can always check both `basic auth` and `with credentials` checkboxes to do the same or am I missing something? Otherwise it just seems strange to me that those 2 settings are tied together.

Reply:
true since we have a separate checkbox for withCredentials we could just check that instead.

Reply:
I assume though that would be a breaking change. I will merge this and create a separate issue for the withCredentials.